### PR TITLE
qa7: Added support of Artix-7 FPGA based board

### PR DESCRIPTION
**Pull request by yshestakov**
Signed-off-by: Yurii Shestakov <yshestakov@gmail.com>

### CODE REVIEW

File: am4/readme.md [hunk 1]
```diff
@@ -103,13 +103,15 @@ The M4 board schematics is updated and Revision 3 seems to be final one.
...
+| DE0     | EP3C16F484C6N   | Cyclone III  | 66 MHz  | 1270 | 491  | 7 M9K   | |
+| DE1     | EP2C20F484C7N   | Cyclone II   | 52 MHz  | 1491 | 477  | 14 M4K  | |
+| DE2-115 | EP4CE115F29C7N  | Cyclone IV   | 60 MHz  | 1598 | 492  | 7 M9K   | |
+| AX309   | XC6SLX9FTG256-2 | Spartan 6    | 64 MHz  | 927  | 464  | 7 BR8   | |
+| DE10-LT | 10M50DAF484C7G  | Max 10       | 58 MHz  | 1195 | 470  | 7 M9K   | |
+| QC5     | 5CEFA2F23I7N    | Cyclone V    | 73 MHz  | 554  | 553  | 6 M10K  | |
+| QC10    | 10CL006U256CN8  | Cyclone 10   | 55 MHz  | 1193 | 471  | 7 M9K   | |
+| EG4     | EG4S20BG256     | Eagle EG4S20 | 59 MHz  | 1021 | 392  | 7 M9K   | |
+| QA7     | XC7A35TFTG256-1 | Artix 7      | n/a     | 631  | 413  | 2 BR | qa7_ram took 2257 LUTs and 139 FF |
+
```
Comment (1801BM1) [inline am4/readme.md:117]: It would be better to set reference with start (*) and add notes below the table

Comment (1801BM1) [reply am4/readme.md:117]: Frequency should be found in timing reports - IIRC, Xilinx reports it as minimal clock period

File: xen/qa7/syn/readme.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+| am4   |   308         |
+
+# Report Instance Areas: 
+## VM1 
+
+	+------+----------------------+-------------+------+
+	|      |Instance              |Module       |Cells |
+	+------+----------------------+-------------+------+
+	|1     |top                   |             |  3182|
+	|2     |  cpu                 |wbc_vm1      |  3109|
```
Comment (1801BM1) [inline xen/qa7/syn/readme.md:34]: The difference: wbc_vm1 - vm1_wb cells looks quite strange. Do we have some explanation?

File: xen/qa7/syn/readme.md [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+	|3     |    cpu               |vm1_wb       |  1760|
+	|4     |      timer           |vm1_timer    |   151|
+	|5     |      vreg_rm         |vm1_reg_ram  |   232|
+	|6     |        vm1_vcram_reg |vm1_vcram    |   226|
+	|7     |    mem               |wbc_mem      |   931|
+	|8     |      ram             |qa7_mem      |   928|
```
Comment (1801BM1) [inline xen/qa7/syn/readme.md:40]: It's a real mess - ~1K LUTS for memory block. Synthesis definitely goes wrong

File: am4/readme.md (hunk 1 above)
Comment (yshestakov) [reply am4/readme.md:117]: Updated Fmax based on WNS value and actual working implementation for Artix-7.

File: xen/qa7/syn/readme.md (hunk 2 above)
Comment (yshestakov) [reply xen/qa7/syn/readme.md:34]: no ideas

File: xen/qa7/syn/readme.md (hunk 3 above)
Comment (yshestakov) [reply xen/qa7/syn/readme.md:40]: It was implemented as "distributed" ram, so seems OK.
I got BRAM based implementation of `wbc_mem`, however I'm not able to get it implemented for 90MHz target, only 85MHz

**Comment by 1801BM1** [comment]
Squashed, updated and merged as:
https://github.com/1801BM1/cpu11/commit/89f85ddd91fda25ed77b15e13868a927737efa1d

**Pull request closed without merging**